tests/test-googledocs-shortcode.php

```diff
@@ -0,0 +1,99 @@
+			'[googledocs url="https://docs.google.com/presentation/d/1tQ4Q1wFpKNLj9BW8s_pCYgDMFXIeHskvTQWaRBS-aGc" height=569 width=960 delayms=3000]'
+		);
+
+		$this->expect_reversal(
+			'<iframe src="https://docs.google.com/presentation/d/1tQ4Q1wFpKNLj9BW8s_pCYgDMFXIeHskvTQWaRBS-aGc/embed?start=true&loop=true&delayms=3000" frameborder="0" width="960" height="569" allowfullscreen="true" mozallowfullscreen="true" webkitallowfullscreen="true"></iframe>',
+			'[googledocs url="https://docs.google.com/presentation/d/1tQ4Q1wFpKNLj9BW8s_pCYgDMFXIeHskvTQWaRBS-aGc" height=569 width=960 start="true" loop="true" delayms=3000]'
+		);
+	}
+
+	public function test_presentation_callback() {
+		$this->expect_callback(
+			'[googledocs url="https://docs.google.com/presentation/d/1tQ4Q1wFpKNLj9BW8s_pCYgDMFXIeHskvTQWaRBS-aGc" delayms=3000 allowfullscreen="true"]',
+			'<iframe class="shortcake-bakery-googledocs-presentation shortcake-bakery-responsive" src="https://docs.google.com/presentation/d/1tQ4Q1wFpKNLj9BW8s_pCYgDMFXIeHskvTQWaRBS-aGc/embed?start=false&loop=false&delayms=3000" frameborder="0" marginheight="0" marginwidth="0" allowfullscreen="true" mozallowfullscreen="true" webkitallowfullscreen="true"></iframe>'
+		);
+	}
+
+	public function test_form_reversal() {
+		$this->expect_reversal(
+			'<iframe src="https://docs.google.com/forms/d/1DwyXsL7kmR2F8-0q0XAecLTwO5_xPeN1tN-ex2Zs_hY/viewform?embedded=true" width="760" height="500" frameborder="0" marginheight="0" marginwidth="0">Loading...</iframe>',
+			'[googledocs url="https://docs.google.com/forms/d/1DwyXsL7kmR2F8-0q0XAecLTwO5_xPeN1tN-ex2Zs_hY" height=500 width=760]'
+		);
+	}
+
+	public function test_form_callback() {
+		$this->expect_callback(
+			'[googledocs url="https://docs.google.com/forms/d/1DwyXsL7kmR2F8-0q0XAecLTwO5_xPeN1tN-ex2Zs_hY"]',
+			'<iframe class="shortcake-bakery-googledocs-form shortcake-bakery-responsive" src="https://docs.google.com/forms/d/1DwyXsL7kmR2F8-0q0XAecLTwO5_xPeN1tN-ex2Zs_hY/viewform?embedded=true" frameborder="0" marginheight="0" marginwidth="0">Loading...</iframe>'
+		);
+	}
+
+	public function test_maps_reversal() {
+		$this->expect_reversal(
+			'<iframe src="https://www.google.com/maps/d/u/1/embed?mid=zEkbFn1A1xVE.kLg_5uTIa64Q" width="640" height="480"></iframe>',
+			'[googledocs url="https://www.google.com/maps/d/embed?mid=zEkbFn1A1xVE.kLg_5uTIa64Q" height=480 width=640]'
+		);
+	}
+
+	public function test_maps_callback() {
+		$this->expect_callback(
+			'[googledocs url="https://www.google.com/maps/d/embed?mid=zEkbFn1A1xVE.kLg_5uTIa64Q"]',
+			'<iframe class="shortcake-bakery-googledocs-map shortcake-bakery-responsive" src="https://www.google.com/maps/d/embed?mid=zEkbFn1A1xVE.kLg_5uTIa64Q" frameborder="0" marginheight="0" marginwidth="0"></iframe>'
+		);
+	}
+
+	private function expect_reversal( $embed, $reversal ) {
```

Comment:
I'm not a huge fan of this abstraction because I think it decreases the readability of the tests. But, not worth changing at this point.